### PR TITLE
Make packer vars overwritable

### PR DESCRIPTION
**Pull request by mrtwnklr**
## Personal goals

I'd like to create a base box using the same localisation settings, similar disk partitioning and similar hostname as my productive instance of Proxmox VE.
Note: All of that might also be possible with a combination of changes in `Vagrantfile` and appropriate shell provisioners.

I'd like to be able to create a customized base box without modifying the original source files of this repository to allow easier recreation of newer base boxes (or: guarantee of `git pull` without conflicts).

## Changes

- `Makefile`: add variable `VAR_FILE` and pass as `-var-file` option to packer
- `upgrade.sh`: modularize into smaller scripts and move them into `provisioners/`
- `proxmox-ve.json`: extract some machine properties, most Proxmox VE installation properties and shell provisioner list to `variables` object to enable override with `-var-file` option

## Tests

Tested with VirtualBox and libvirt only as I have no HyperV setup.
Testing with libvirt was a bit strange: Proxmox VE is not able to detect and prefill the country field. When executed with VirtualBox it does. So you should not set `step_country` to empty when using libvirt. I modified the documented sample in the `README.md`.

* `make build-[virtualbox|libvirt]`
    * **OK:** default values from `proxmox-ve.json:variables` are entered correctly (verified with `headless: false` while typing `boot_command`)
    * **OK:** `vagrant up --provider=[virtualbox|libvirt]` is successfull
    * **OK:** Proxmox VE is reachable via browser
* `make build-[virtualbox|libvirt] VAR_FILE=example-test.vars.json`, with file `example-test.vars.json`:

    ```json
    {
        "cpus": "4",
        "disk_size": "131072",
        "memory": "8192",
        "step_country": "Ger<wait>m<wait>a<wait>n<wait><enter>",
        "step_keyboard_layout": "<end><up><wait>",
        "step_email": "example-test@example.com",
        "step_hostname": "example-test.local",
        "shell_provisioner_scripts": "provisioners/upgrade.sh,provisioners/network.sh,localisation-de.sh,provisioners/reboot.sh,provisioners/provision.sh"
    }
    ```

    * **OK:** given values from ``example-test.vars.json`` are entered correctly (verified with `headless: false` while typing `boot_command`)
    * **OK:** german localisation is applied (with custom `localisation-de.sh`)
    * **OK:** `vagrant up --provider=[virtualbox|libvirt]` is successfull
    * **OK:** Proxmox VE is reachable via browser

Versions used:

```bash
$ packer --version
1.7.9

$ vagrant --version
Vagrant 2.2.19

$ qemu-system-x86_64 --version
QEMU emulator version 4.2.1 (Debian 1:4.2-3ubuntu6.19)

$ vboxmanage --version
6.1.26_Ubuntur145957
```

### CODE REVIEW

**Comment by mrtwnklr** [comment]
@rgl What do you think about this pr?

**Comment by rgl** [comment]
@mrtwnklr, it looks good!

PS I'm sorry, but I forgot about this.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+VAR_FILE:=
```
Comment (rgl) [inline Makefile:1]: can you please add a space? to be like `VAR_FILE :=`

File: README.md [hunk 2]
```diff
@@ -83,6 +83,39 @@ cd example
...
+* sets the country to Germany (timezone is updated by Proxmox VE installer) and changes
+  the keyboard layout back to "U.S. English" as this is needed for the subsequent
+  `boot_command` statements
+* uses all default shell provisioners (see [`./provisioners`](./provisioners)) and a
+  custom one for german localisation
+
+  ```json
+  {
+      "disk_size": "131072",
+      "memory": "4048",
```
Comment (rgl) [inline README.md:112]: this should be `4096`?

File: README.md [hunk 3]
```diff
@@ -83,6 +83,39 @@ cd example
...
+      "step_country": "Ger<wait>m<wait>a<wait>n<wait><enter>",
+      "step_keyboard_layout": "<end><up><wait>",
+      "shell_provisioner_scripts": "provisioners/upgrade.sh,provisioners/network.sh,localisation-de.sh,provisioners/reboot.sh,provisioners/provision.sh"
+  }
+  ```
+
 # Packer boot_command
```
Comment (rgl) [inline README.md:119]: what do you think about updating this `# Packer boot_command` section to use these `step_` variables?

**Comment by mrtwnklr** [comment]
> @mrtwnklr, it looks good!
> 
> PS I'm sorry, but I forgot about this.

No problem. I'll continue the work on this PR after #19 is merged to reduce noise with merge conflicts.

File: README.md (hunk 3 above)
Comment (mrtwnklr) [reply README.md:119]: Ok. What do you have in mind?

```markdown
| step                              | boot_command                                          |
|----------------------------------:|-------------------------------------------------------|
| [...]                             |                                                       |
| type country                      | ```{{user `step_country`}}<tab><wait>```              |
| timezone                          | ```{{user `step_timezone`}}<tab><wait>```             |
| keyboard layout                   | ```{{user `step_keyboard_layout`}}<tab><wait>```      |
| [...]                             |                                                       |
| email                             | ```{{user `step_email`}}<tab><wait>```                |
| [...]                             |                                                       |
| hostname                          | ```{{user `step_hostname`}}<tab><wait>```             |
| [...]                             |                                                       |
```

**Comment by mrtwnklr** [comment]
Done the rebase and rework.

I retested it only with libvirt as I have no Virtualbox or HyperV installed.

File: README.md (hunk 3 above)
Comment (rgl) [reply README.md:154]: yes, just like that. does it make sense?

Comment (mrtwnklr) [reply README.md:154]: For me, as I know the code, this indirection is ok.

But as a new user I would prefer to read the concrete values instead.

Comment (rgl) [reply README.md:154]: Makes sense. Its good as-is. I will test, merge this.

**Comment by rgl** [comment]
Merged. Thank you!

**Pull request merged**